drivers/i2c/i2c_bitbang.h

```diff
@@ -48,6 +48,12 @@ void i2c_bitbang_init(struct i2c_bitbang *bitbang,
  */
 int i2c_bitbang_configure(struct i2c_bitbang *bitbang, u32_t dev_config);
 
+/**
+ * Implementation of the functionality required by the 'recover_bus'
+ * function in struct i2c_driver_api.
+ */
+int i2c_bitbang_recover_bus(struct i2c_bitbang *bitbang);
```

Comment:
To follow up on my API comment: my concern about visibility here is that if a driver (e.g. Nordic) wants to make use of this single function, it's still including a header and module that provides a bunch of structures and functions that provide significantly more functionality that's not necessary.

Yes if not referenced the linker will likely discard the code, and presumably there won't be any harm in having the extra functions, but it's still getting more stuff than is necessary to the intended task.